provider/provider.go

```diff
@@ -128,19 +128,33 @@ func (p *dockerNativeProvider) Check(ctx context.Context, req *rpc.CheckRequest)
 	logging.V(9).Infof("%s executing", label)
 
 	inputs, err := plugin.UnmarshalProperties(req.GetNews(), plugin.MarshalOptions{
-		KeepUnknowns: false,
+		KeepUnknowns: true,
 		SkipNulls:    true,
 	})
 	if err != nil {
 		return nil, err
 	}
 
+	if inputs["build"].ContainsUnknowns() {
+		// We skip some of the "nice-to-have" default and verification logic in the case of unknowns.
+		// This should be fine, given that _any_ unknowns in the Build field should trigger a diff.
+		// Furthermore, all of this will get called again during `pulumi up`.
+		inputStruct, err := plugin.MarshalProperties(inputs, plugin.MarshalOptions{
+			KeepUnknowns: false,
```

Comment:
Why `KeepUnknowns: false,`? This means that `create/update(inputs, preview=true)` will see `inputs["build"]` as empty instead of unknown. This is probably fine since we have disabled preview, but I'm still curious why the choice.

If it doesn't matter, we can just omit setting `KeepUnknowns`.

Reply:
True but mitigated by preview=true is not being called.

Reply:
So this short-circuits not only verification logic but also default application logic, all of it. have we thought that through? Is that perhaps wanted? Like this is short-circuited `build.Dockerfile = path.Join(build.Context, defaultDockerfile)` . Any unknowns at all in the "build" input would short circuit default application even when individual values are known to be empty.

Reply:
I guess this bit is wanted "This means our custom foo.Dockerfile is not being used and docker tries to use the default Dockerfile resulting in the following error:"